### PR TITLE
SAAS-3762: Added DMS Publicly Accessible Instances Plugin

### PR DESCRIPTION
**Pull request by sadeed12345**

### CODE REVIEW

File: plugins/aws/dms/dmsPubliclyAccessibleInstances.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+module.exports = {
+    title: 'DMS Publicly Accessible Instances',
+    category: 'DMS',
+    domain: 'Application Integration',
+    description: 'Ensure that in order to avoid exposing sensitive data your Amazon Database Migration Service (DMS) are not publicly accessible.',
+    more_info: 'An AWS DMS replication instance can have one public IP address and one private IP address. If you uncheck (disable) the box for Publicly accessible, then the replication instance has only a private IP address. that prevents from exposure of data to other users',
+    recommended_action: 'Disable Publicly Accessible feature in order to have private IP address and avoid exposing of private data.',
+    link: 'https://docs.aws.amazon.com/dms/latest/userguide/CHAP_ReplicationInstance.PublicPrivate.html',
+    apis: ['DMS:describeReplicationInstances'],
```
Comment (m-nouman-suleman) [inline plugins/aws/dms/dmsPubliclyAccessibleInstances.js:12]: Please remove this

File: plugins/aws/dms/dmsPubliclyAccessibleInstances.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'DMS Publicly Accessible Instances',
+    category: 'DMS',
+    domain: 'Application Integration',
+    description: 'Ensure that in order to avoid exposing sensitive data your Amazon Database Migration Service (DMS) are not publicly accessible.',
```
Comment (AkhtarAmir) [inline plugins/aws/dms/dmsPubliclyAccessibleInstances.js:8]:
```suggestion
    description: 'Ensure that Amazon Database Migration Service (DMS) instances are not publicly accessible.',
```

File: plugins/aws/dms/dmsPubliclyAccessibleInstances.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                    'No DMS replication instances found', region);
+                return rcb();
+            }
+
+            for (let instance of describeReplicationInstances.data) {
+                if (!instance.ReplicationInstanceArn) continue;
+
+                let resource = instance.ReplicationInstanceArn;
+
+                if (instance.PubliclyAccessible == false) {
```
Comment (AkhtarAmir) [inline plugins/aws/dms/dmsPubliclyAccessibleInstances.js:42]:
```suggestion
                if (!instance.PubliclyAccessible) {
```

File: plugins/aws/dms/dmsPubliclyAccessibleInstances.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+                let resource = instance.ReplicationInstanceArn;
+
+                if (!instance.PubliclyAccessible) {
+                    helpers.addResult(results, 0,
+                        'DMS replication instances are not publicly accessible.',
+                        region, resource);
+                } else {
+                    helpers.addResult(results, 2,
+                        'DMS replication instances are publicly accessible.',
```
Comment (AkhtarAmir) [inline plugins/aws/dms/dmsPubliclyAccessibleInstances.js:48]:
```suggestion
                        'DMS replication instance is not publicly accessible.',
                        region, resource);
                } else {
                    helpers.addResult(results, 2,
                        'DMS replication instance is publicly accessible.',
```

File: plugins/aws/dms/dmsPubliclyAccessibleInstances.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'DMS Publicly Accessible Instances',
+    category: 'DMS',
+    domain: 'Application Integration',
+    description: 'Ensure that Amazon Database Migration Service (DMS) instances are not publicly accessible.',
+    more_info: 'An AWS DMS replication instance can have one public IP address and one private IP address. If you uncheck (disable) the box for Publicly accessible, then the replication instance has only a private IP address. that prevents from exposure of data to other users',
+    recommended_action: 'Disable Publicly Accessible feature in order to have private IP address and avoid exposing of private data.',
```
Comment (AkhtarAmir) [inline plugins/aws/dms/dmsPubliclyAccessibleInstances.js:10]:
```suggestion
    recommended_action: 'Ensure that DMS replication instances have only private IP address and not public IP address',
```

**Review by AkhtarAmir — APPROVED** [review]

**Review by dipsubha06 — APPROVED** [review]

**Pull request merged**